prog/rand.go

```diff
@@ -337,12 +338,20 @@ func (r *randGen) randFromMap(m map[string]bool) string {
 
 func (r *randGen) randString(s *state, t *BufferType) []byte {
 	if len(t.Values) != 0 {
-		return []byte(t.Values[r.Intn(len(t.Values))])
+		res := []byte(t.Values[r.Intn(len(t.Values))])
+		if r.currCall != nil && r.currCall.Meta.Attrs.KFuzzTest && t.Kind == BufferString {
+			res = append(res, 0)
```

Comment:
1. Move the comment to the first place where you do patching.
2. You might also need to patch string mutation.

Ideally, you should have patched specifically the `if r.oneOf(100) == t.NoZ` condition here. The `t.Values` case is irrelevant for you unless you add enums (and if we don't null terminate enum strings, that'd be a bug in syzkaller).